src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private User user1 = new User("spider@gmail.com", "qwerty12345", 25);
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_AlreadyExistedUser_notOk() {
+        registrationService.register(user1);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user1));
+    }
+
+    @Test
+    public void register_nullUser_notOk() {
+        user1 = null;
+        assertThrows(NullPointerException.class, () -> registrationService.register(user1));
+    }
+
+    @Test
+    public void register_adult_notOk() {
+        user1.setAge(17);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user1));
+    }
```

Comment:
In this case what will be other user1 fields?  null? or "spider..." and "qwerty..."?